op-batcher/readme.md

```diff
@@ -29,23 +29,52 @@ The philosophy behind the current architecture is:
 * When something goes wrong, we rewind the state cursors by the minimal amount we need to get going again.
 
 
-### Happy path
+### Routines
 
-In the happy path, the batcher periodically:
-0. Queries the sequencer's syncStatus and
+The batcher has up to 4 concurrent routines:
+
+The `readLoop`, which (in the happy path)
+1. Queries the sequencer's syncStatus and
   a. (optionally) waits for it to ingest more L1 data before taking action
   b. prunes blocks and channels from its internal state which are no longer required
-1. Enqueues unsafe blocks and dequeues safe blocks from the sequencer to its internal state.
+2. Enqueues unsafe blocks and dequeues safe blocks from the sequencer to the batcher's internal state.
+3. Signals to the `writeLoop` and to the `throttlingLoop` when new blocks are loaded
+
+The `writeLoop` which
+1. Waits for a signal from the `readLoop`
 2. Enqueues a new channel, if necessary.
 3. Processes some unprocessed blocks into the current channel, triggers the compression of the block data and the creation of frames.
 4. Sends frames from the channel queue to the DA layer as (e.g. to Ethereum L1 as calldata or blob transactions).
-5. If there is more transaction data to send, go to 2. Else wait for a tick and go to 0.
-
+5. If there is more transaction data to send, go to 2. Else go to 1.
+
+The `receiptsLoop` which
+1. Receives receipts from handlers spawned by the `writeLoop`.
+2. Updates metrics
+3. (Conditionally) Modifies the batcher state to cope when a channel times out on chain (see below).
+
+The `throttlingLoop` which
+1. Waits for a signal from the `readLoop`
+2. If it detects that the pending data in state is over a threshold, calls the sequencer over RPC and tells it to throttle the amount of L2 data it produces. See the (section below)[#data-availability-backlog]
+
+The relationships are shown in this diagram:
+```mermaid
+architecture-beta
+    group bs(server)[BatchSubmitter]
+
+    service readLoop(server)[readLoop] in bs
+    service writeLoop(server)[writeLoop] in bs
```

Comment:
I have a suggestion to rename the loops to make it clearer what they do, and avoid the confusion why the readLoop is producing data for two other loops:
* `readLoop` -> `loaderLoop` or `blockLoaderLoop` or so
* `writeLoop` -> `publishLoop` or so

Also in code of course, not only here

Reply:
Yeah those names work and are in better harmony with existing nomenclature in this component.

Reply:
[7196f19](https://github.com/ethereum-optimism/optimism/pull/14244/commits/7196f1919c30780251887b50253a6f4644dfde04)